module/input/NatNet/src/NatNet.cpp

```diff
@@ -379,17 +388,15 @@ namespace module::input {
         }
     }
 
-    void NatNet::processPing(const Packet& packet) {
+    void NatNet::process_ping(const Packet& packet) {
 
         // Extract the information from the packet
         std::string name(&packet.data);
         const char* app_version     = &packet.data + 256;
         const char* nat_net_version = app_version + 4;
 
         // Update our version number
-        version = (nat_net_version[0] << 24) | (nat_net_version[1] << 16) | (nat_net_version[2] << 8)
-                  | (nat_net_version[3] << 0);
-
+        version = ntohl(*reinterpret_cast<const char*>(nat_net_version));
```

Comment:
To use ntohl (network to host long) you need to use a 32 bit integer, so you'd need to do the cast to one. This will just cast it to a char and then try to do it for that (you'll probably miss most of your version number)

so `ntohl(*reinterpret_cast<const uint32_t*>(nat_net_version));`